Conflicting versions of Netty

## Problem
Using the `zeebe-client-java` and `zeebe-test` in version 0.19.0 can lead to the following ClassNotFoundException:

```
SEVERE: [Channel<4>: (0.0.0.0:25600)] Uncaught exception in the SynchronizationContext. Panic!
java.lang.NoClassDefFoundError: io/netty/util/internal/ReferenceCountUpdater
	at io.netty.buffer.Unpooled.wrappedBuffer(Unpooled.java:157)
	at io.netty.handler.codec.http2.Http2ConnectionHandler.<clinit>(Http2ConnectionHandler.java:74)
	at io.grpc.netty.NettyClientTransport.start(NettyClientTransport.java:205)
	at io.grpc.internal.ForwardingConnectionClientTransport.start(ForwardingConnectionClientTransport.java:33)
	at io.grpc.internal.ForwardingConnectionClientTransport.start(ForwardingConnectionClientTransport.java:33)
	at io.grpc.internal.InternalSubchannel.startNewTransport(InternalSubchannel.java:271)
	at io.grpc.internal.InternalSubchannel.obtainActiveTransport(InternalSubchannel.java:216)
	at io.grpc.internal.ManagedChannelImpl$SubchannelImpl.requestConnection(ManagedChannelImpl.java:1508)
	at io.grpc.internal.PickFirstLoadBalancer.handleResolvedAddresses(PickFirstLoadBalancer.java:59)
	at io.grpc.internal.AutoConfiguredLoadBalancerFactory$AutoConfiguredLoadBalancer.handleResolvedAddresses(AutoConfiguredLoadBalancerFactory.java:155)
	at io.grpc.internal.ManagedChannelImpl$NameResolverListener$1NamesResolved.run(ManagedChannelImpl.java:1378)
	at io.grpc.SynchronizationContext.drain(SynchronizationContext.java:88)
	at io.grpc.SynchronizationContext.execute(SynchronizationContext.java:120)
	at io.grpc.internal.ManagedChannelImpl$NameResolverListener.onResult(ManagedChannelImpl.java:1388)
	at io.grpc.internal.DnsNameResolver$Resolve.resolveInternal(DnsNameResolver.java:311)
	at io.grpc.internal.DnsNameResolver$Resolve.run(DnsNameResolver.java:213)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: io.netty.util.internal.ReferenceCountUpdater
```

The problem is caused by different versions of Netty libraries. 
* `io.grpc:grpc-netty:jar:1.21.0` -> `io.netty:netty-codec-http2:jar:4.1.34.Final`
* `io.zeebe:broker-core` -> `io.netty:netty-codec-http:jar:4.1.36.Final`
* `io.zeebe:atomix-cluster` -> `io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.27.Final`

## Possible Solution
* upgrade to `grpc-netty` version (1.22.0) when it's released
* downgrade version in `broker-core` to 4.1.34.Final

## Workaround
In addition to `zeebe-bom`, also include the `netty-bom`.

```xml
<dependencyManagement>
		<dependencies>
			<dependency>
				<groupId>io.zeebe</groupId>
				<artifactId>zeebe-bom</artifactId>
				<version>0.19.0-alpha3</version>
				<scope>import</scope>
				<type>pom</type>
			</dependency>
			<dependency>
				<groupId>io.netty</groupId>
				<artifactId>netty-bom</artifactId>
				<version>4.1.37.Final</version>
				<scope>import</scope>
				<type>pom</type>
			</dependency>
		</dependencies>
	</dependencyManagement>
```

Well, the error has changed to

```java
java.util.concurrent.CompletionException: java.lang.NoSuchMethodError: io.netty.util.concurrent.SingleThreadEventExecutor.<init>(Lio/netty/util/concurrent/EventExecutorGroup;Ljava/util/concurrent/Executor;ZLjava/util/Queue;Lio/netty/util/concurrent/RejectedExecutionHandler;)V
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273) ~[?:1.8.0_212]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280) ~[?:1.8.0_212]
	at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:961) ~[?:1.8.0_212]
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:926) ~[?:1.8.0_212]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:442) ~[?:1.8.0_212]
	at io.atomix.utils.concurrent.SingleThreadContext$1.lambda$execute$0(SingleThreadContext.java:53) ~[atomix-utils-3.2.0-alpha2.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_212]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_212]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_212]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[?:1.8.0_212]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_212]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_212]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_212]
Caused by: java.lang.NoSuchMethodError: io.netty.util.concurrent.SingleThreadEventExecutor.<init>(Lio/netty/util/concurrent/EventExecutorGroup;Ljava/util/concurrent/Executor;ZLjava/util/Queue;Lio/netty/util/concurrent/RejectedExecutionHandler;)V
	at io.netty.channel.SingleThreadEventLoop.<init>(SingleThreadEventLoop.java:65) ~[netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.channel.nio.NioEventLoop.<init>(NioEventLoop.java:137) ~[netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.channel.nio.NioEventLoopGroup.newChild(NioEventLoopGroup.java:138) ~[netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.channel.nio.NioEventLoopGroup.newChild(NioEventLoopGroup.java:37) ~[netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:84) ~[netty-common-4.1.34.Final.jar:4.1.34.Final]
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:58) ~[netty-common-4.1.34.Final.jar:4.1.34.Final]
	at io.netty.util.concurrent.MultithreadEventExecutorGroup.<init>(MultithreadEventExecutorGroup.java:47) ~[netty-common-4.1.34.Final.jar:4.1.34.Final]
	at io.netty.channel.MultithreadEventLoopGroup.<init>(MultithreadEventLoopGroup.java:59) ~[netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:78) ~[netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:73) ~[netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.channel.nio.NioEventLoopGroup.<init>(NioEventLoopGroup.java:60) ~[netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.atomix.cluster.messaging.impl.NettyUnicastService.start(NettyUnicastService.java:181) ~[atomix-cluster-3.2.0-alpha2.jar:?]
	at io.atomix.cluster.AtomixCluster.lambda$startServices$2(AtomixCluster.java:300) ~[atomix-cluster-3.2.0-alpha2.jar:?]
	at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:952) ~[?:1.8.0_212]
	... 10 more
```

It seems that we need to upgrade the netty version also in Atomix :balloon:

--- a/bom/pom.xml
+++ b/bom/pom.xml
@@ -24,6 +24,8 @@
     </nexus.snapshot.repository>
     <nexus.release.repository>https://app.camunda.com/nexus/content/repositories/zeebe-io/
     </nexus.release.repository>
+
+    <version.netty>4.1.37.Final</version.netty>
   </properties>
 
   <dependencyManagement>
@@ -176,6 +178,14 @@
         <groupId>io.zeebe</groupId>
         <artifactId>zeebe-protocol-asserts</artifactId>
         <version>${project.version}</version>
+      </dependency>
+
+      <dependency>
+        <groupId>io.netty</groupId>
+        <artifactId>netty-bom</artifactId>
+        <version>${version.netty}</version>
+        <type>pom</type>
+        <scope>import</scope>
       </dependency>
     </dependencies>
   </dependencyManagement>

--- a/parent/pom.xml
+++ b/parent/pom.xml
@@ -324,14 +324,6 @@
       </dependency>
 
       <dependency>
-        <groupId>io.netty</groupId>
-        <artifactId>netty-bom</artifactId>
-        <version>${version.netty}</version>
-        <type>pom</type>
-        <scope>import</scope>
-      </dependency>
-
-      <dependency>
         <groupId>io.grpc</groupId>
         <artifactId>grpc-netty</artifactId>
         <version>${version.grpc}</version>

